api4/config.go

```diff
@@ -79,7 +79,7 @@ func updateConfig(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 
 	appCfg := c.App.Config()
-	if *appCfg.ServiceSettings.SiteURL != "" && *cfg.ServiceSettings.SiteURL == "" {
+	if *appCfg.ServiceSettings.SiteURL != "" && (cfg.ServiceSettings.SiteURL == nil || *cfg.ServiceSettings.SiteURL == "") {
```

Comment:
How is it that `cfg.ServiceSettings.SiteURL` can be `nil` here? On line 74 we call `cfg.SetDefaults` which calls `ServiceSettings.SetDefaults` which ensures `SiteURL` wont be nil.

Reply:
I forgot to update the method and saw the test fail but you are right, only the patch has the issue. Let me update.